src/Avalonia.Base/Utilities/SpanHelpers.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+
+namespace Avalonia.Utilities
+{
+    public static class SpanHelpers
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool TryParseFromHexToUInt(this ReadOnlySpan<char> span, out uint value)
+        {
+#if NETSTANDARD2_0
+            return uint.TryParse(span.ToString(), NumberStyles.HexNumber, CultureInfo.InvariantCulture,
+                    out value);
+#else
+            return uint.TryParse(span, NumberStyles.HexNumber, CultureInfo.InvariantCulture,
+                    out value);
+#endif
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool TryParseToInt(this ReadOnlySpan<char> span, out int value)
+        {
+#if NETSTANDARD2_0
+            return int.TryParse(span.ToString(), out value);
+#else
+            return int.TryParse(span, out value);
+#endif
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool TryParseNumberToDouble(this ReadOnlySpan<char> span, out double value)
+        {
+#if NETSTANDARD2_0
+            return double.TryParse(span.ToString(), NumberStyles.Number, CultureInfo.InvariantCulture,
+                    out value);
+#else
+            return double.TryParse(span, NumberStyles.Number, CultureInfo.InvariantCulture,
```

Comment:
This is really the correct fix. Honestly, I would copy-paste this method out of newer .net versions and just add it Avalonia. Then we can use it uniformly in new code.

Reply:
https://github.com/dotnet/runtime/blob/4574ccb53981aaecfbf5e514a5114da2321e0843/src/libraries/System.Private.CoreLib/src/System/Number.Parsing.cs#L2439-L2498

Reply:
Yes, the point was add methods to fallback to in .net standard 2.0. MathUtilities was just an example. Different direction is fine though since it's already merged.